### PR TITLE
Use `babelrc: false` only for custom config in `.storybook` directory

### LINKED ISSUE #2582: Support for `.babelrc.js`
### Issue details

Currently `@storybook/react` only looks for Babel configuration in `.babelrc` file:

https://github.com/storybooks/storybook/blob/d63bf65afa75f2f6f4c0f63b0fedb990a8ea98d0/app/react/src/server/babel_config.js#L48

In `@babel/core` they support more configuration sources (`.babelrc.js`, `.babelrc`, `package.json`):

https://github.com/babel/babel/blob/7d798952d2a305cf77d24d1707f723f0a162aadd/packages/babel-core/src/config/files/configuration.js#L29

Looks like the functionality of reading Babel configuration is common enough to be extracted into its own package. I guess it's not good enough just to reimplement the same logic in Storybook repo. Or we can just use Babel's internals – though it will be not easy to maintain down the road.

Any thoughts?

### Steps to reproduce

1. Have Babel configuration in `.babelrc.js` file.
2. Run `@storybook/react` with something like `start-storybook -p 9001 -c .storybook`.
3. `.babelrc.js` is ignored

Expected behaviour: Storybook reads Babel configuration from `.babelrc.js` file.

### Please specify which version of Storybook and optionally any affected addons that you're running

- @storybook/react 3.3.1

### Affected platforms 

babel-loader 7.1.2
babel-core ^7.0.0-0
@babel/core 7.0.0-beta.34

### PR DESCRIPTION
**Pull request by Hypnosphi**
Fixes  #2582 #2771 #2610 #754 #2635 

This also removes the need to duplicate our presets (react, env etc) in user's app babel config

### CODE REVIEW

File: lib/core/src/server/loadCustomBabelConfig.js [hunk 1]
```diff
@@ -43,24 +45,25 @@ function loadFromPath(babelConfigPath) {
...
-      babelConfig.extends = inConfigDir
-        ? path.resolve(configDir, babelConfig.extends)
-        : path.resolve(babelConfig.extends);
+      babelConfig.extends = path.resolve(configDir, babelConfig.extends);
     }
+    return babelConfig;
   }
 
-  return babelConfig || defaultConfig;
+  return isBabelLoader8() ? defaultConfig : {};
```
Comment (igor-dv) [inline lib/core/src/server/loadCustomBabelConfig.js]: So, our default babel config is for babel8. After removing `babelrc: false`, can't we just return the default?

**Comment by igor-dv** [comment]
BTW, by

> This also removes the need to duplicate our presets (react, env etc) in user's app babel config

Can we get rid of `wrapDefaultBabelConfig` ?

File: lib/core/src/server/loadCustomBabelConfig.js (hunk 1 above)
Comment (igor-dv) [reply lib/core/src/server/loadCustomBabelConfig.js]: Ah, just understood my mistake 🤦‍♂️

File: examples/marko-cli/package.json [hunk 2]
```diff
@@ -19,8 +19,7 @@
   },
   "dependencies": {
     "marko": "^4.12.4",
-    "marko-starter": "^2.0.4",
-    "marko-widgets": "^7.0.1"
```
Comment (ndelangen) [inline examples/marko-cli/package.json]: why delete the widgets?

File: examples/marko-cli/src/stories/addon-knobs.stories.js [hunk 3]
```diff
@@ -17,27 +15,3 @@ storiesOf('Addons|Knobs/Hello', module)
       age,
     });
   });
-
-storiesOf('Addons|Knobs/Hello (Marko Widget)', module)
```
[the remark this answers is not in the document]
Comment (Hypnosphi) [reply examples/marko-cli/src/stories/addon-knobs.stories.js]: Looks like they decided to drop this functionality

File: examples/marko-cli/package.json (hunk 2 above)
Comment (Hypnosphi) [reply examples/marko-cli/package.json]: They are deprecated and don't support marko 4

**Comment by Hypnosphi** [comment]
> You mean we can get rid of wrapDefaultBabelConfig ?

No, it means that our default config is now being merged with user's

File: examples/marko-cli/package.json (hunk 2 above)
Comment (ndelangen) [reply examples/marko-cli/package.json]: okay, clear!

**Review by ndelangen — APPROVED** [review]
This is great, this makes so much sense!

**Pull request merged**